src/lib/optimizer/strategy/abstract_rule.cpp

```diff
@@ -9,9 +9,17 @@
 
 namespace hyrise {
 
-void AbstractRule::apply_to_plan(const std::shared_ptr<LogicalPlanRootNode>& lqp_root) const {
+IsCacheable operator&(IsCacheable lhs, IsCacheable rhs) {
+  return static_cast<IsCacheable>(static_cast<bool>(lhs) & static_cast<bool>(rhs));
+}
+
+IsCacheable& operator&=(IsCacheable& lhs, const IsCacheable rhs) {
```

Comment:
out of curiosity: why is only `IsCacheable` parameter passed as reference?

Reply:
I'm not quite sure if I get the question right: `lhs` is passed by reference, because we actually want to modify this parameter with the operator. We don't want to modify `rhs`, so this parameter is passed by value.
But I guess, the return type of the function can be `IsCacheable` instead of `IsCacheable&`. Good catch!

Reply:
Actually, the explanation regarding the `lhs` is correct, but returning a reference is just the normal way to implement that operator: returning the result of the assignment again instead of making a copy.